EnglishLanguageArts/PartsOfSpeech/parts-of-speech.ipynb

```diff
@@ -0,0 +1,249 @@
+    "    he.append(0)\n",
+    "    she.append(0)\n",
+    "    me.append(0)\n",
+    "    I.append(0)\n",
+    "    for token in nlp(frankenstein['Chapter Text'][row_number]):\n",
+    "        if token.lower_ == 'he':\n",
+    "            he[row_number] += 1\n",
+    "        if token.lower_ == 'she':\n",
+    "            she[row_number] += 1\n",
+    "        if token.lower_ == 'me':\n",
+    "            me[row_number] += 1\n",
+    "        if token.text == 'I':\n",
+    "           I[row_number] += 1\n",
+    "frankenstein['he'] = he\n",
+    "frankenstein['she'] = she\n",
+    "frankenstein['me'] = me\n",
+    "frankenstein['I'] = I\n",
+    "frankenstein.drop(columns=['Chapter Text','Length']).cumsum().plot(\n",
+    "    title='Cumulative Pronouns in Frankenstein',labels=dict(index='Chapter'))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Anne of Green Gables\n",
+    "by Lucy Maud Montgomery\n",
+    "\n",
+    "### Parts of Speech"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "url = 'http://www.gutenberg.org/files/45/45-0.txt'\n",
+    "r = requests.get(url)\n",
+    "r.encoding = 'utf-8'\n",
+    "text = r.text.split('***')[2]\n",
+    "text = text.replace(\"’\",\"'\").replace(\"“\",'\"').replace(\"”\",'\"')\n",
+    "\n",
+    "anne = pd.DataFrame()\n",
+    "for chapter in text.split('CHAPTER'):\n",
+    "    if len(chapter)>500:\n",
+    "        chapter = chapter.replace('\\r',' ').replace('\\n',' ').replace('  ', ' ')\n",
+    "        anne = anne.append({'Chapter Text':chapter, 'Length':len(chapter)}, ignore_index=True)\n",
+    "word_types = ['ADJ', 'ADV', 'VERB', 'NOUN', 'PRON', 'PROPN'] # https://spacy.io/api/annotation#pos-tagging\n",
+    "pos_df = pd.DataFrame(columns=word_types)\n",
+    "for i in range(len(anne)):\n",
+    "    pos_list = []\n",
+    "    for token in nlp(anne['Chapter Text'][i]):\n",
+    "        pos = token.pos_\n",
+    "        if pos in word_types:\n",
+    "            pos_list.append(pos)\n",
+    "    word_type_count = {}\n",
+    "    for word_type in word_types:\n",
+    "        word_type_count.update({word_type:pos_list.count(word_type)/len(anne['Chapter Text'][i])})\n",
+    "    pos_df = pos_df.append(word_type_count, ignore_index=True)\n",
+    "pos_df['Chapter'] = pos_df.index+1\n",
+    "pos_df = pos_df.set_index('Chapter')\n",
+    "pos_df.plot.line(title='Parts of Speech Proportions in Anne of Green Gables by Chapter',labels=dict(value='Proportion of Total Words'))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Pronoun Usage"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "he = []\n",
+    "she = []\n",
+    "me = []\n",
+    "I = []\n",
+    "for row_number in range(len(anne)):\n",
```

Comment:
Since this takes a while to run it would be good to have a print statement like previously
print('Running natural language processing, please wait...')